client/src/common/types/Semester.ts

```diff
@@ -4,6 +4,10 @@ import type Course from "./Course";
  * Semester type, contains a list of courses
  */
 interface Semester {
+    term: string;
```

Comment:
Make this an number, I'd rather the seasons be represented as ints, like 0 for spring, 1 for summer, 2 for fall, 3 for winter